src/main/java/appeng/tile/storage/TileChest.java

```diff
@@ -259,37 +263,38 @@ else if( fluidCell != null )
 			}
 		}
 
-		switch( channel )
+		if( channel == AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) )
 		{
-			case FLUIDS:
-				if( this.fluidCell == null )
-				{
-					throw NO_HANDLER;
-				}
-				return this.fluidCell;
-			case ITEMS:
-				if( this.itemCell == null )
-				{
-					throw NO_HANDLER;
-				}
-				return this.itemCell;
-			default:
+			if( this.itemCell == null )
+			{
+				throw NO_HANDLER;
+			}
+			return this.itemCell;
+		}
+		else if( channel == AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) && this.fluidCell != null )
+		{
+			if( this.fluidCell == null )
+			{
+				throw NO_HANDLER;
+			}
+			return this.fluidCell;
+
 		}
 
 		return null;
 	}
 
-	private <StackType extends IAEStack> MEMonitorHandler<StackType> wrap( final IMEInventoryHandler h )
+	private <T extends IAEStack<T>> MEMonitorHandler<T> wrap( final IMEInventoryHandler<T> h )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "MEMonitorHandler"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)